### PR TITLE
Update flake8 to 7.1.1.

### PR DESCRIPTION
**Pull request by bdice**
## Description
We need to update flake8 to fix a false-positive that appears with older flake8 versions on Python 3.12.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by mroeschke — APPROVED** [review]

**Review by jameslamb — APPROVED** [review]

File: python/cuspatial/cuspatial/tests/test_geodataframe.py [hunk 1]
```diff
@@ -93,8 +93,8 @@ def assert_eq_multipolygon(p1, p2):
 
 
 def assert_eq_geo_df(geo1, geo2):
-    if type(geo1) != type(geo2):
-        assert TypeError
+    if type(geo1) is not type(geo2):
+        raise TypeError
```
Comment (bdice) [inline python/cuspatial/cuspatial/tests/test_geodataframe.py:97]:
```suggestion
    assert type(geo1) is type(geo2)
```

[the remark this answers is not in the document]
Comment (jameslamb) [reply python/cuspatial/cuspatial/tests/test_geodataframe.py]: I wanted to help so looked into this a bit.

Here's the specific error I now see in the one failing test:

```text
FAILED tests/test_geodataframe.py::test_cudf_dataframe_init - AssertionError: assert <class 'geopandas.geodataframe.GeoDataFrame'> is <class 'pandas.core.frame.DataFrame'>
 +  where <class 'geopandas.geodataframe.GeoDataFrame'> = type(   a  b\n0  1  4\n1  2  5\n2  3  6)
 +  and   <class 'pandas.core.frame.DataFrame'> = type(   a  b\n0  1  4\n1  2  5\n2  3  6)
```

([build link](https://github.com/rapidsai/cuspatial/actions/runs/10910352860/job/30281407991?pr=1458#step:8:274))

Here's the source code of that test:

https://github.com/rapidsai/cuspatial/blob/a2d21c94bc5e53bca030f6bb3603bd7ac2c770dd/python/cuspatial/cuspatial/tests/test_geodataframe.py#L462-L465

It looks to me like it's intentional behavior that the `cuspatial.GeoDataFrame.to_pandas()` method actually returns a `geopandas.GeoDataFrame`, not a `pandas.DataFrame`.

https://github.com/rapidsai/cuspatial/blob/a2d21c94bc5e53bca030f6bb3603bd7ac2c770dd/python/cuspatial/cuspatial/core/geodataframe.py#L70-L80

So I guess one of these has to happen:

* `cuspatial.GeoDataFrame.to_pandas()` needs to start returning a `pandas.DataFrame` instead (as the name implies)
* the have-the-same-type expectation needs to be removed from that test

[the remark this answers is not in the document]
Comment (mroeschke) [reply python/cuspatial/cuspatial/tests/test_geodataframe.py]: From a cursory check, it appears the intention is for this check to assert that both arguments are `geopandas.GeoDataFrame` objects, so I think the usage here is incorrect.

I imagine the test should have been `assert_eq_geo_df(gdf.to_pandas(), gpd.GeoDataFrame(df.to_pandas()))`?

[the remark this answers is not in the document]
Comment (bdice) [reply python/cuspatial/cuspatial/tests/test_geodataframe.py]: I'll try that locally. Thanks, that sounds like the right direction to me.

[the remark this answers is not in the document]
Comment (bdice) [reply python/cuspatial/cuspatial/tests/test_geodataframe.py]: Yup, that worked on my local build. I pushed the change. Thanks @mroeschke!

**Pull request merged**